docs/metrics/llm-prompt-metric.md

```diff
@@ -0,0 +1,174 @@
+Then, on a new line, explain the rating in 10 words.
+```
+
+```py
+response2 = client.chat.completions.create(
+    model="gpt-4-1106-preview",
+    messages=[
+      {
+        "role": "user",
+        "content": INITIAL_PROMPT
+      },
+      {
+        "role": "assistant",
+        "content": "Certainly! Please provide me with the pair of texts for evaluation."
+      },
+      {
+        "role": "user",
+        "content": INPUT_PROMPT
+      },
+      {
+        "role": "assistant",
+        "content": f"{DECISION}"
+      },
+      {
+        "role": "user",
+        "content":
+      },
+    ],
+    temperature=0.5,
+    max_tokens=256,
+    top_p=1
+)
+```
+
+After some post-processing, the output will be something similar to below:
+```py
+(
+    'Yes', # Are the pair of texts hallucinating?
+    'The second text contradicts the first about digestibility.', # Explanation for why
+    '0.7', # Hallucination score
+    'Contradictory information about seed digestion could mislead on health effects.' # Explanation of score
+)
+```
+
+### Chain-of-Thought (CoT) Prompt
+
+Prompt engineering with CoT is similar to the aforementioned two phase approach with GPT-4, however,
+our initial prompt includes a zero-shot chain-of-thought technique that works simply by adding in a
+phrase like "Let's think step by step."
+
+An example of such a prompt would be:
+
+```
+In the context of NLP, a 'hallucination' refers to a phenomenon where the LLM generates text that is incorrect, nonsensical, or not real.
+Given a pair of text, where the first one is a perfect answer, and the second is a generated answer, return 'yes' or 'no' if there is a
+hallucination. If yes, on the next line, explain in about ten words why there is a hallucination. Otherwise, simply return 'no'.
+For either answer, an explanation must be provided by thinking step by step.
+```
+
+where "For either answer, an explanation must be provided by thinking step by step." is the only
+new addition to the previous prompt `INITIAL_PROMPT`.
+
+"[Chain-of-Thought Prompting Elicits Reasoning in Large Language Models](https://arxiv.org/abs/2201.11903)" by Wei et al.
+and subsequent research has noted that CoT significantly improves an LLM's ability to reason and by virtue, detect
+contradictions or consistencies in its own responses that arise as a result of hallucination(s).
+
+### Self-Consistency
+
+Self-consistency is the act of repeating the the same query `k` times and aggregating across the sample responses to
+obtain the most consistent answer. This can help us improve the accuracy of most prompting techniques mentioned above.
+
+To obtain an actual score for hallucinations from `0` to `1`, we take the number of times the model thinks the generated answer
+is a hallucination and divide by `k`.
+
+## Limitations and Advantages
+
+1. **Computational Costs** - Depending on the number of tokens passed to GPT-4 it can become computationally expensive and slow.
```

Comment:
Since we are using an API model, can we really say it's "computationally expensive"?  I think it should be just "costly" as you are paying based on the # of tokens.